src/schemes/boundary/dummy_particles/dummy_particles.jl

```diff
@@ -378,68 +415,115 @@ function compute_pressure!(boundary_model, ::Union{PressureMirroring, PressureZe
     return boundary_model
 end
 
-@inline function adami_pressure_extrapolation_neighbor!(boundary_model, system,
-                                                        neighbor_system, system_coords,
-                                                        neighbor_coords, v_neighbor_system,
-                                                        neighborhood_search)
+@inline function dynamic_pressure(density_neighbor, v, v_neighbor_system, particle,
+                                  neighbor, system::BoundarySystem)
+    if system.ismoving[]
+        return 0.5 * density_neighbor *
+               norm(current_velocity(v, system, particle) -
+                    v_neighbor_system[1:ndims(system), neighbor])^2
+    end
+    return 0.0
+end
```

Comment:
```suggestion
@inline function dynamic_pressure(density_neighbor, v, v_neighbor, particle,
                                  neighbor, system::BoundarySystem, neighbor_system)
    if system.ismoving[]
        relative_velocity = current_velocity(v, system, particle) - current_velocity(v_neighbor, neighbor_system, neighbor)
        return density_neighbor * dot(relative_velocity, relative_velocity) / 2
    end
    return zero(density_neighbor)
end
```
Several remarks here:
1. Never assume that `v_neighbor_system` just stores the velocity in the first dimensions. That's what `current_velocity` is for.
2. Slicing an array causes allocations. Again, this is what `current_velocity` is for.
3. `dot(x, x)` is about 2x faster than `norm(x)`.
4. Use `/ 2` and `zero(density_neighbor)` to make sure that everything stays `Float32` when you input `Float32`.
5. Can you please move this function down below `bnd_pressure_inner!(boundary_model, ::BernoulliPressureExtrapolation, ...)`?